pangeo/Pystac_Basics.ipynb

```diff
@@ -0,0 +1,2315 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "220919ef-d7e7-4c92-9171-9f23fe25b84b",
+   "metadata": {},
+   "source": [
+    "# Using STAC Basics\n",
+    " \n",
+    "In this notebook, we utilise basic functions to explore the SentinelHub Catalog using Pystac."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "0d8352d5-6bb1-40c3-9ce8-ddc71c202cf1",
+   "metadata": {},
+   "source": [
+    "For running the examples in this Jupyter Notebook you will need to install the pystac-client. To install it from PyPI using pip, use the following command:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "id": "b4db954d-69eb-4bb6-a71c-66370889f98a",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "!pip install pystac_client"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "996c25de-d18f-4965-b901-626ec4c56aa0",
+   "metadata": {},
+   "source": [
+    "In order to access the functionalities of the client API, we should import the pystac_client package along with other necessary packages as follows:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "id": "17cdffbf-eb1c-4c52-bfd0-cd0812298567",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from pystac import Catalog\n",
+    "from pystac import StacIO\n",
+    "\n",
+    "from pystac_client import Client\n",
+    "from odc.stac import stac_load, configure_rio\n",
+    "import geopandas as gpd\n",
+    "import folium, shapely"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "d7f71c6e-b0f3-45d4-a8b2-e32c43b66f5c",
+   "metadata": {},
+   "source": [
+    "The Sentinel Hub API uses OAuth2 Authentication and requires that we have an access token. Thus in python the requests-oauthlib library can handle the retrieval of access tokens using our OAuth Client configuration."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "id": "942e390a-3b13-48aa-86fd-f8345331070b",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def get_auth():\n",
```

Comment:
This `get_auth()` snippet introduces another case of bad auth credentials/secret handling

also see https://github.com/eu-cdse/notebook-samples/issues/13 and https://github.com/eu-cdse/documentation/issues/191

Reply:
used getpass based method as a solution.